src/indicator/algorithm/calculator/stochasticOscillator.js

```diff
@@ -0,0 +1,68 @@
+import d3 from 'd3';
+import _slidingWindow from './slidingWindow';
+import {rebind} from '../../../util/rebind';
+
+export default function() {
+
+    var closeValue = function(d, i) { return d.close; },
+        highValue = function(d, i) { return d.high; },
+        lowValue = function(d, i) { return d.low; };
+
+    var kWindow = _slidingWindow()
+        .windowSize(5)
+        .accumulator(function(values) {
+            var maxHigh = d3.max(values, highValue);
+            var minLow = d3.min(values, lowValue);
+            return 100 * (closeValue(values[values.length - 1]) - minLow) / (maxHigh - minLow);
+        });
+
+    var dWindow = _slidingWindow()
+        .windowSize(3)
+        .accumulator(function(values) {
+            if (values[0] === undefined) {
+                return undefined;
+            }
+            return d3.mean(values);
+        });
+
+    var stochastic = function(data) {
+        var kValues = kWindow(data);
+        var dValues = dWindow(kValues);
+        return kValues.map(function(k, i) {
```

Comment:
Not sure which difference you mean? They sound the same based on these descriptions -

[underscore](http://underscorejs.org/#zip)
>Merges together the values of each of the arrays with the values at the corresponding position. Useful when you have separate data sources that are coordinated through matching array indexes. If you're working with a matrix of nested arrays, _.zip.apply can transpose the matrix in a similar fashion.
```js
_.zip(['moe', 'larry', 'curly'], [30, 40, 50], [true, false, false]);
=> [["moe", 30, true], ["larry", 40, false], ["curly", 50, false]]
```

[d3](https://github.com/mbostock/d3/wiki/Arrays#d3_zip)
>Returns an array of arrays, where the ith array contains the ith element from each of the argument arrays. The returned array is truncated in length to the shortest array in arrays. If arrays contains only a single array, the returned array contains one-element arrays. With no arguments, the returned array is empty.
```js
d3.zip([1, 2], [3, 4]); // returns [[1, 3], [2, 4]]
```